jsk_tools/src/audible_warning.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import actionlib
+import rospy
+from threading import Event
+from threading import Thread
+from threading import Lock
+from sound_play.msg import SoundRequest
+from sound_play.msg import SoundRequestAction
+from sound_play.msg import SoundRequestGoal
+from diagnostic_msgs.msg import DiagnosticArray
+from diagnostic_msgs.msg import DiagnosticStatus
+
+cached_paths = {}
+cached_result = {}
+
+
+def is_leaf(original_path):
+    if hasattr(cached_result, original_path):
+        return cached_result[original_path]
+
+    path = original_path[1:].split('/')
+
+    def _is_leaf(cached, path):
+        if not isinstance(path, list):
+            raise ValueError
+        if len(path) == 0:
+            raise ValueError
+        if len(path) == 1:
+            if path[0] in cached:
+                return False
+            else:
+                return True
+
+        parent = path[0]
+        child = path[1:]
+        if parent not in cached:
+            cached[parent] = {}
+        return _is_leaf(cached[parent], child)
+
+    result = _is_leaf(cached_paths, path)
+    if result is False:
+        cached_result[original_path] = result
+    return result
+
+
+class SpeakThread(Thread):
+
+    def __init__(self, rate=1.0, wait=True, blacklist=None):
+        super(SpeakThread, self).__init__()
+        self.memo = {}
+        self.event = Event()
+        self.rate = rate
+        self.wait = wait
+        self.lock = Lock()
+        self.stale = []
+        self.error = []
+        self.history = defaultdict(lambda: 10)
+        self.blacklist = blacklist
+
+        self.talk = actionlib.SimpleActionClient(
+            "/robotsound", SoundRequestAction)
+        self.talk.wait_for_server()
+
+        self.language = 'en'
+
+    def stop(self):
+        self.event.set()
+
+    def sort(self, error):
+        ret = {}
+        for s in error:
+            ns = s.name.split()[0]
```

Comment:
Because of this line,
`s = "/Motors/EtherCAT Device (br_caster_l_wheel_motor)"`
becomes
`ns = "/Motors/EtherCAT"`

Therefore, these PR2 warnings
```
/Motors/EtherCAT Device (br_caster_r_wheel_motor)
/Motors/EtherCAT Device (br_caster_l_wheel_motor)
/Motors/EtherCAT Device flr_caster_r_wheel_motor)
...
```
become the same name.

![audible_warning](https://user-images.githubusercontent.com/19769486/168327101-4cd5375f-53bf-4aeb-8c5f-d325383d8c0c.png)

We need to Need to come up with a name matching system for blacklist.